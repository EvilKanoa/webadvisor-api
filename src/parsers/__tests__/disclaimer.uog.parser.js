const path = require('path');
const util = require('util');
const readFile = util.promisify(require('fs').readFile);
const { parseDisclaimer } = require('../disclaimer.uog.parser');

describe('disclaimer.uog.parser', () => {
  it('handles empty inputs', async () => {
    const empty = { title: undefined, description: undefined };
    expect(await parseDisclaimer('')).toEqual(empty);
    expect(await parseDisclaimer(null)).toEqual(empty);
    expect(await parseDisclaimer(false)).toEqual(empty);
    expect(await parseDisclaimer('<html></html>')).toEqual(empty);
  });

  it('parses a disclaimer correctlys', async () => {
    const html = await readFile(
      path.join(__dirname, '__data__', 'disclaimer.uog.data.1.html'),
    );
    const result = await parseDisclaimer(html);
    const str = util.inspect(result, {
      depth: 10,
      maxArrayLength: Infinity,
      sorted: true,
    });
    expect(str).toMatchSnapshot();
  });
});
