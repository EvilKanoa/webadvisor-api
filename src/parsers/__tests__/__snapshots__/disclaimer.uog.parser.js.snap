// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disclaimer.uog.parser parses a disclaimer correctlys 1`] = `
"{ description:
   'The information published in this Undergraduate Calendar outlines the rules, regulations, curricula, programs and fees for the 2018-2019 academic year, including the Summer Semester 2018, the Fall Semester 2018 and the Winter Semester 2019.\\\\nThe University reserves the right to change without notice any information contained in this calendar, including fees, any rule or regulation pertaining to the standards for admission to, the requirements for the continuation of study in, and the requirements for the granting of degrees or diplomas in any or all of its programs. The publication of information in this calendar does not bind the University to the provision of courses, programs, schedules of studies, or facilities as listed herein.\\\\nThe University will not be liable for any interruption in, or cancellation of, any academic activities as set forth in this calendar and related information where such interruption is caused by fire, strike, lock-out, inability to procure materials or trades, restrictive laws or governmental regulations, actions taken by faculty, staff or students of the University or by others, civil unrest or disobedience, public health emergencies, or any other cause of any kind beyond the reasonable control of the University.\\\\nIn the event of a discrepancy between a print version (downloaded) and the Web version, the Web version will apply,\\\\nPublished by: Enrolment Services',
  title: 'Disclaimer: University of Guelph 2018' }"
`;
